src/pipdeptree/_discovery.py

```diff
@@ -1,35 +1,39 @@
 from __future__ import annotations
 
+import sys
+from importlib.metadata import distributions
+from pathlib import Path
 from typing import TYPE_CHECKING
 
 if TYPE_CHECKING:
-    from pip._vendor.pkg_resources import DistInfoDistribution
+    from importlib.metadata import Distribution
+
+
+def is_virtual_environment():
+    return hasattr(sys, "prefix") and sys.prefix != sys.base_prefix
```

Comment:
The following would be better as it comes straight from the Python [docs](https://docs.python.org/3/library/venv.html#how-venvs-work) (add this URL link to as a comment for better clarity):
```suggestion
def is_virtual_environment():
    return sys.prefix != sys.base_prefix
```

Reply:
Also make sure that you add type hints to your functions.

Reply:
fixed.